src/main/webapp/app/overview/discussion-section/discussion-section.component.ts

```diff
@@ -154,12 +162,73 @@ export class DiscussionSectionComponent extends CourseDiscussionDirective implem
         return this.sortByDate(postA, postB, SortDirection.ASCENDING);
     };
 
+    /**
+     * Set the channel for the discussion section, either for a lecture or an exercise
+     * @param courseId
+     */
+    setChannel(courseId: number): void {
+        if (this.course?.courseInformationSharingConfiguration === CourseInformationSharingConfiguration.COMMUNICATION_ONLY) {
+            this.metisService.getFilteredPosts({
+                exerciseId: this.exercise?.id,
+                lectureId: this.lecture?.id,
+            });
+            return;
+        }
+        const getChannel = () => {
+            return {
+                next: (channel: Channel) => {
+                    this.channel = channel ?? undefined;
+
+                    if (!this.channel && this.course?.courseInformationSharingConfiguration === CourseInformationSharingConfiguration.MESSAGING_ONLY) {
+                        this.noChannelAvailable = true;
+                        this.collapsed = true;
+                        return;
+                    }
+
+                    const contextFilter = this.channel?.id
+                        ? { conversationId: this.channel.id }
+                        : {
+                              exerciseId: this.exercise?.id,
+                              lectureId: this.lecture?.id,
+                          };
+                    this.metisService.getFilteredPosts(contextFilter);
+                    this.createEmptyPost();
+                    this.resetFormGroup();
+                },
+                error: (error: HttpErrorResponse) => {
+                    if (error.status === 403 && error.error.message === 'error.noAccessButCouldJoin') {
+                        this.isNotAChannelMember = true;
+                        this.collapsed = true;
+                    }
+                },
+            };
+        };
+
+        if (this.lecture?.id) {
+            this.channelService
+                .getChannelOfLecture(courseId, this.lecture.id)
+                .pipe(map((res: HttpResponse<Channel>) => res.body))
+                .subscribe(getChannel());
+        } else if (this.exercise?.id) {
+            this.channelService
+                .getChannelOfExercise(courseId, this.exercise.id)
+                .pipe(map((res: HttpResponse<Channel>) => res.body))
+                .subscribe(getChannel());
+        }
```

Comment:
I'm wondering if we can avoid these additional REST calls by including them in a previous request aka the ones where the lecture or the exercise come from? 🤔